src/connection.rs

```diff
@@ -1176,46 +1192,30 @@ where
 
         trace!("Buffering frame to {}:\n{}", self.peer_addr(), frame);
 
-        frame.format(self.out_buffer.get_mut())?;
+        frame.format(&mut self.out_buffer)?;
         Ok(())
     }
 
     fn check_buffer_out(&mut self, frame: &Frame) -> Result<()> {
-        if self.out_buffer.get_ref().remaining() < frame.len() {
-            // There is no more room to grow, and we can't shift the buffer
-            if self.out_buffer.position() == 0 {
-                return Err(Error::new(
-                    Kind::Capacity,
-                    "Reached the limit of the output buffer for the connection.",
-                ));
-            }
-
-            // Shift the buffer
-            let prev_pos = self.out_buffer.position() as usize;
-            self.out_buffer.set_position(0);
-            self.out_buffer.get_mut().shift(prev_pos);
+        if self.out_buffer.remaining_mut() < frame.len() {
```

Comment:
I'm not sure `self.out_buffer.remaining_mut() < frame.len()` should be `self.out_buffer.remaining_mut() > frame.len()` (or `>=`)?

it looks `self.out_buffer.remaining_mut()` means `The maximum amount of bytes that can still be written to this buffer.`  and `frame` should be written to the `out_buffer`